solr/benchmark/src/java/org/apache/solr/bench/lifecycle/SolrStartup.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.bench.lifecycle;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.lucene.util.IOUtils;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.request.CoreAdminRequest;
+import org.apache.solr.client.solrj.response.CoreAdminResponse;
+import org.apache.solr.embedded.JettyConfig;
+import org.apache.solr.embedded.JettySolrRunner;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+
+/**
+ * A simple JMH benchmark that attempts to measure approximate Solr startup behavior by measuring
+ * {@link JettySolrRunner#start()}
+ */
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Measurement(time = 60, iterations = 10)
+@Threads(3)
+@Fork(value = 1)
+public class SolrStartup {
+
+  @Benchmark
+  public void startSolr(PerThreadState threadState) throws Exception {
+    threadState.solrRunner.start(
+        false); // 'false' tells Jetty to not go out of its way to reuse any previous ports
+  }
+
+  @State(Scope.Thread)
+  public static class PerThreadState {
+
+    private static final int NUM_CORES = 10;
+
+    public File tmpSolrHome;
```

Comment:
I think it simplifies things to leave `tmpSolrHome` as a `Path` and not convert it to a file?

There are things like `PathUtils` in commons-io if needed

Reply:
IMO it's a bit of a wash either way.  JettySolrRunner and other objects have ctors and methods that require a string or File, so even with the change there's an ugly amount of object conversion.  Boy, I just hate working with files in Java.

But, "done".

Reply:
@madrob had made quite a bit of progress converting from File -> Path. One example is https://issues.apache.org/jira/browse/SOLR-15919 so trying not to go backwards.